### PR TITLE
fix: Strict Validation Rules greater_than/less_than

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
From https://github.com/codeigniter4/CodeIgniter4/issues/6489#issuecomment-1236288297

- fix type error in `greater_than` `greater_than_equal_to` `less_than` `less_than_equal_to`
  - `CodeIgniter\Validation\Rules::greater_than_equal_to(): Argument https://github.com/codeigniter4/CodeIgniter4/pull/1 ($str) must be of type ?string, int given, called in /var/www/src/system/Validation/StrictRules/Rules.php on line 88`
- refactor StrictRules/RulesTest

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Validation/StrictRules/Rules.php [hunk 1]
```diff
@@ -71,20 +71,28 @@ public function exact_length($str, string $val): bool
...
      * Greater than
      *
-     * @param mixed $str
+     * @param mixed $str expects int|string
      */
     public function greater_than($str, string $min): bool
     {
+        if (is_int($str)) {
+            $str = (string) $str;
+        }
```
Comment (MGatner) [inline system/Validation/StrictRules/Rules.php:80]: Is there a problem with using `@param int|string $str` and `$str = (string) $str;`?

Comment (kenjis) [reply system/Validation/StrictRules/Rules.php:80]: `$str` is user input, and we don't know what type it is.

But I found another bug. I'll fix.

Comment (kenjis) [reply system/Validation/StrictRules/Rules.php:80]: `$str = (string) $str;` makes `true` `'1'`.
Then `true` will be greater_than `0`. So it is not good.

I pushed a commit to fix another TypeError.

Comment (MGatner) [reply system/Validation/StrictRules/Rules.php:80]: The cast is dependent on the docblock type. At this point in dev in 2022 I'm fine requiring users to adhere to docblock types; if we say the parameter is `int|string` and someone passes `bool` then that's on them for not implementing any of the static analysis tools we and the larger PHP community provide.

This works though and is more conservative so it's fine to leave it how you have it.

**Review by MGatner — APPROVED** [review]

File: system/Validation/StrictRules/Rules.php (hunk 1 above)
Comment (kenjis) [reply system/Validation/StrictRules/Rules.php:80]: A site user may send unexpected typed values.
In that case, if it is not an attack, returning false is better than TypeError.

Validation rules are special cases. Devs cannot force the input type.
So we cannot set `int|string` in this case.

**Pull request merged**